core-java-modules/core-java-compiler/src/main/java/com/baeldung/compilerApi/JavaCompilerUtils.java

```diff
@@ -0,0 +1,134 @@
+package com.baeldung.compilerApi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.tools.*;
+import java.io.*;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * A utility class for compiling Java source code using the Java Compiler API.
+ * This class provides methods to compile Java code from strings or files.
+ */
```

Comment:
I was also advised by the team that we have guidelines to not use comments to add explanations for the code. All the required explanations should be in the text. Can you please remove the comments and JavaDoc from the PR to address this, and move any comments that are important to note into the article text instead.